clients/drcachesim/tracer/output.cpp

```diff
@@ -1059,8 +1079,11 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
     BUF_PTR(data->seg_base) += append_unit_header(drcontext, BUF_PTR(data->seg_base),
                                                   dr_get_thread_id(drcontext), window);
     num_refs_racy += current_num_refs;
-    if (op_exit_after_tracing.get_value() > 0 &&
-        num_refs_racy > op_exit_after_tracing.get_value()) {
+    if (mode == BBDUP_MODE_L0_FILTER) {
+        num_filter_refs_racy += current_num_refs;
+    }
+    if (mode != BBDUP_MODE_L0_FILTER && op_exit_after_tracing.get_value() > 0 &&
+        (num_refs_racy - num_filter_refs_racy) > op_exit_after_tracing.get_value()) {
```

Comment:
Instead of tracking num_filter_refs_racy and using `num_refs_racy - num_filter_refs_racy` everywhere, can we simply set num_refs_racy to zero near the similar `data->bytes_written = 0` at L1018?

Reply:
Yeah I thought of doing that but then I also thought that if we printed that number in some debug logs, then it might be confusing to see the number of references go down.

Reply:
I left a comment on the `data->bytes_written = 0;` line. There might be an issue with reseting that. So perhaps it is better to keep this separate `num_filter_refs_racy`.